validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorConfig.java

```
@@ -472,6 +484,11 @@ public Builder doppelgangerDetectionEnabled(final boolean doppelgangerDetectionE
       return this;
     }
 
+    public Builder runnerThreadNum(final int runnerThreadNum) {
+      this.runnerThreadNum = runnerThreadNum;
```

Might be worth validating here that we don't do something silly, even though it's a develop option...
it'd just be something like 
```
if (executorThreads < 1 || executorThreads > 5_000) {
throw new invalidConfigurationException("Invalid configuration. --Xvalidator-client-executor-threads must be greater than 0 and less than 5000.")
}
```
I picked 5000 arbitrarily, it could easily limit to something more like 1000 from your test, but something above what you found useful in your situation...
I guess the main thing is that if someone goes and puts a very large number, it may cause other issues.
Note also the check > 0 is only really window dressing, as it will be caught by the max logic. It's nicer that the user knows they've done something silly so I added it in...
The result of this validation would be an exception that stops the client running early (fail fast) so that they can correct the issue and get into a running state...

---

Yeah this makes sense. I have added a check like you suggested and a corresponding test. Might make sense for consistency to do this with the `--Xrest-api-validator-threads` option at some point too.

---

Fair, I think I might have missed that PR, it's something I'd normally suggest to be a little user tolerant and fail fast :)